web-app/src/main/java/com/google/sps/servlets/AnalyticsServlet.java

```diff
@@ -54,4 +55,17 @@ public void init() {
       }
     }
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Coordinates[] waypoints;
+    Gson gson = new Gson();
+
+    try {
+      JsonReader reader = new JsonReader(request.getReader());
+      waypoints = new Gson().fromJson(reader, Coordinates[].class);
+    } catch (Exception e) {
+      System.out.println(e);
```

Comment:
nit: Instead of system.out maybe writing exception to response allows to deal error in client side when no analysis was returned from server?